test-suite-groovy/src/test/groovy/io/micronaut/docs/aop/retry/BookService.groovy

```diff
@@ -0,0 +1,50 @@
+package io.micronaut.docs.aop.retry
+
+import io.micronaut.retry.annotation.CircuitBreaker
+import io.micronaut.retry.annotation.Retryable
+import io.reactivex.Flowable
+
+class BookService {
+
+    // tag::simple[]
+    @Retryable
+    List<Book> listBooks() {
+        // ...
+    // end::simple[]
+        [new Book("The Stand")]
+    }
+
+    // tag::circuit[]
+    @CircuitBreaker(reset = "30s")
+    List<Book> findBooks() {
+        // ...
+    // end::circuit[]
+        [new Book("The Stand")]
+    }
+
+    // tag::attempts[]
+    @Retryable( attempts = "5",
+                delay = "2s" )
+    Book findBook(String title) {
+        // ...
+    // end::attempts[]
+        new Book(title)
+    }
+
+    // tag::config[]
+    @Retryable( attempts = "\${book.retry.attempts:3}",
```

Comment:
Did I do this right for injecting config values in here? I spent a long time trying to figure out what "${...}" does in Java but I couldn't turn anything up, so I took the auto-converter at its word that this should be a literal string that contains a g-string, and I'm guessing it gets read as such somewhere in annotation processing?

Reply:
It gets parsed at runtime. It is a literal string, but it does not contain a GString. If the `$` was not escaped then it would be a GString and it would fail.